types/react/experimental.d.ts

```diff
@@ -38,7 +38,24 @@ import React = require('.');
 
 export {};
 
+type NonNullableValue = string | number | boolean | object | symbol;
```

Comment:
Why can't these be `null` or `undefined`?

Reply:
Because I borrowed them from React flowtype definition. They don't seem to expect to use the maybe type here. 

See here for the implementation detail: https://github.com/facebook/react/commit/322cdcd3abfaca985a001a12247f02c5d31d311e#diff-1adf25deacc6168be0ac290f98561486R961-R962

Reply:
I haven't tested yet if maybe type doesn't cause problems. I'll test it soon

Reply:
Passing undefined or null doesn't throw an internal exception.

But, should we allow it? I don't think it makes a good use case.

Reply:
Alright, I was just curious since the PR had no links to docs, types or source.

It seems like that `Exclude<unknown, null | undefined>` would (theoratically) be equivalent to `$NonMaybeType<mixed>`. [This doesn't work](https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=9&pc=1#code/C4TwDgpgBAJAcgewHYFkCGIBGEAq4IoCWAHhACZQC8UAosQMYA2ArmRADzNIDWSCA7kgA0UJM0aMoAHyhc2AM0JJyAPgBQa+V3rBCyKAGUEzAE70IACgDOxsxABcsRKgzY8kIqTIBKKAG8oAF8NAHoQqAABYCsAWghiSB04kxMEEzUjU3MLMQlvAG41MMjouISIJIgUtIzbbLkIRWUfQqA) but it might be good to add a comment explaining what we're trying to achieve with NonNullableValue ("faking `$NonMaybeType<mixed>`").